OrbitCore/SymbolHelper.cpp

```diff
@@ -137,3 +161,36 @@ ErrorMessageOr<ModuleSymbols> SymbolHelper::LoadUsingSymbolsPathFile(
     const std::string& module_path, const std::string& build_id) const {
   return FindSymbols(module_path, build_id, symbols_file_directories_);
 }
+
+ErrorMessageOr<std::string> SymbolHelper::FindDebugSymbolsFile(
+    const std::string& module_path, const std::string& build_id) const {
+  return FindSymbolsFile(module_path, collector_symbol_directories_, build_id);
+}
+
+ErrorMessageOr<ModuleSymbols> SymbolHelper::LoadSymbolsFromFile(
+    const std::string& file_path, const std::string& build_id) const {
+  ErrorMessageOr<std::unique_ptr<ElfFile>> elf_file_result =
+      ElfFile::Create(file_path);
+
+  if (!elf_file_result) {
+    return ErrorMessage(
+        absl::StrFormat("Failed to load debug symbols from \"%s\": %s",
+                        file_path, elf_file_result.error().message()));
+  }
+
+  const std::string& target_build_id = elf_file_result.value()->GetBuildId();
+  if (target_build_id != build_id) {
```

Comment:
could both of these strings be empty? 

Consider an elf file that contains symbols but not a build id. This might actually become a problem at other code locations as well

Reply:
that's a good point.

Reply:
added this use-case and a test for this.